geode-core/src/main/java/org/apache/geode/cache/query/internal/QueryConfigurationServiceImpl.java

```diff
@@ -0,0 +1,176 @@
+  private static final String NO_CLASS_FOUND =
+      "No MethodInvocationAuthorizer class found with name ";
+  private static final String NO_VALID_CONSTRUCTOR =
+      "No valid MethodInvocationAuthorizer constructor available. ";
+  private static final String INSTANTIATION_ERROR =
+      "Error occurred while instantiating MethodInvocationAuthorizer. ";
+  private static final String AUTHORIZER_NOT_UPDATED = "The authorizer was not updated.";
+
+  private MethodInvocationAuthorizer authorizer;
+
+  /**
+   * Instead of the below property, please use the UnrestrictedMethodAuthorizer
+   * implementation of MethodInvocationAuthorizer, which provides the same functionality.
+   */
+  @Deprecated
+  public final boolean ALLOW_UNTRUSTED_METHOD_INVOCATION;
+
+  public static final String DEPRECATION_WARNING = "The property " + GEMFIRE_PREFIX +
+      "QueryService.allowUntrustedMethodInvocation is deprecated. " +
+      "Please use the UnrestrictedMethodAuthorizer implementation of MethodInvocationAuthorizer " +
+      "instead";
+
+  public QueryConfigurationServiceImpl() {
+    ALLOW_UNTRUSTED_METHOD_INVOCATION = Boolean.parseBoolean(
+        System.getProperty(GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation"));
+  }
+
+  public static MethodInvocationAuthorizer getNoOpAuthorizer() {
+    // A no-op authorizer, allow method invocation
+    return ((Method m, Object t) -> true);
+  }
+
+  @Override
+  public boolean init(Cache cache) {
+    if (cache == null) {
+      throw new IllegalArgumentException("cache must not be null");
+    }
+
+    if (System
+        .getProperty(GEMFIRE_PREFIX + "QueryService.allowUntrustedMethodInvocation") != null) {
+      logger.warn(DEPRECATION_WARNING);
+    }
+
+    if (isSecurityDisabled((InternalCache) cache) || ALLOW_UNTRUSTED_METHOD_INVOCATION) {
+      this.authorizer = getNoOpAuthorizer();
+    } else {
+      this.authorizer = new RestrictedMethodAuthorizer(cache);
+    }
+    return true;
+  }
+
+  @Override
+  public Class<? extends CacheService> getInterface() {
+    return QueryConfigurationService.class;
+  }
+
+  @Override
+  public CacheServiceMBeanBase getMBean() {
+    return null;
+  }
+
+  @Override
+  public void close() {
+
+  }
+
+  @Override
+  public MethodInvocationAuthorizer getMethodAuthorizer() {
+    return authorizer;
+  }
+
+  @Override
+  public void updateMethodAuthorizer(Cache cache, QueryMethodAuthorizerCreation creation) {
+    updateMethodAuthorizer(cache, creation.getClassName(), creation.getParameters());
+  }
+
+  @Override
+  public void updateMethodAuthorizer(Cache cache, String className, Set<String> parameters) {
+    if (isSecurityDisabled((InternalCache) cache) || ALLOW_UNTRUSTED_METHOD_INVOCATION) {
+      return;
+    }
+
+    if (className == null) {
+      logError(UPDATE_ERROR_MESSAGE + NULL_CLASS_NAME + AUTHORIZER_NOT_UPDATED);
+      return;
+    }
+
+    if (className.equals(RestrictedMethodAuthorizer.class.getSimpleName())) {
```

Comment:
is there a reason why we use simple name instead of the entire path?  If we use simple name, we probably need to document that these are 'reserved' in that if a customer tries to create their own authorizers with the same name, they will end up getting our prebuilt ones?

Reply:
We can definitely switch to using the full name for the out-of-the-box authorizers. The choice to use simple names was just one of convenience, to prevent the user having to type the whole thing out, but it definitely saves confusion to use the same pattern in all cases.